.github/workflows/testing-dependency.yaml

```diff
@@ -96,8 +91,58 @@ jobs:
           PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY }}
           INDEX_NAME: ${{ needs.dependency-matrix-setup.outputs.index_name }}  
 
-
-
+  dependency-matrix-grpc-312:
+    name: Deps (GRPC)
+    needs: dependency-matrix-setup
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 5
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.11
+          - 3.12
+        grpcio-version:
+          - 1.59.3
+          - 1.60.0
+        lz4-version:
+          - 3.1.3
+          - 4.3.3
+        protobuf-version:
+          - 3.20.3
+        googleapis-common-protos-version:
+          - 1.53.0
+          - 1.62.0
+        grpc-gateway-protoc-gen-openapiv2-version:
+          - 0.1.0
+    steps:
```

Comment:
These test steps are the same as above, but unfortunately github actions doesn't support yaml anchors that would enable easy reuse in file https://github.com/actions/runner/issues/1182